isisd/isis_tlvs.c

```diff
@@ -4723,6 +4960,16 @@ void isis_tlvs_set_te_router_id(struct isis_tlvs *tlvs,
 	memcpy(tlvs->te_router_id, id, sizeof(*id));
 }
 
+void isis_tlvs_set_te_router_id_ipv6(struct isis_tlvs *tlvs,
```

Comment:
This function seems to be assuming that new value will always be different from old value. In case that's not true we will be unnecessarily freeing and re-allocating memory. Please ensure that's not a possible scenario.

Reply:
In fact, the function handle both case: removal and modification. Just have a look to the function above that deal with IPv4 router ID. I just wrote the same for IPv6.